src/messaging.cc

```diff
@@ -186,6 +186,10 @@ bool MessageEncoder::encode_any(Object* object) {
       word from = Smi::value(from_object);
       word to = Smi::value(to_object);
       if (is_array(list)) return encode_array(Array::cast(list), from, to);
+      Smi* perhaps_list_class_id = list->class_id();
+      if (perhaps_list_class_id != program->list_class_id()) {
+        return false;
+      }
```

Comment:
I'd move this up to right after you fetch the `list` from the slice (line 182).

```suggestion
      // We currently do not support serializing slices of non-default lists.
      if (list->class_id() != program->list_class_id()) return false;
```